core/src/main/kotlin/org/evomaster/core/search/gene/GeneUtils.kt

```diff
@@ -138,50 +158,63 @@ object GeneUtils {
 
      */
 
-    fun applyEscapes(string: String, mode: String = "none", format: OutputFormat = OutputFormat.KOTLIN_JUNIT_5): String{
+    fun applyEscapes(string: String, mode: EscapeMode = EscapeMode.NONE, format: OutputFormat): String{
         val ret = when (mode){
-            "uris" -> applyUriEscapes(string, format)
-            "queries" -> applyQueryEscapes(string, format)
-            "assertions" -> applyAssertionEscapes(string, format)
-            "json" -> applyJsonEscapes(string, format)
-            "text" -> applyTextEscapes(string, format)
-            else -> string
+            EscapeMode.URI -> applyUriEscapes(string, format)
+            EscapeMode.SQL -> applySqlEscapes(string, format)
+            EscapeMode.ASSERTION -> applyAssertionEscapes(string, format)
+            EscapeMode.JSON -> applyJsonEscapes(string, format)
+            EscapeMode.TEXT -> applyTextEscapes(string, format)
+            EscapeMode.NONE -> string
         }
         //if(forQueries) return applyQueryEscapes(string, format)
         //else return applyAssertionEscapes(string, format)
         return ret
     }
 
-    fun applyJsonEscapes(string: String, format: OutputFormat = OutputFormat.JAVA_JUNIT_4):String{
-        val ret = when{
-            format.isKotlin() -> string.replace("\$", "\\\$")
-            else -> string
-        }
-        return ret
-                .replace("\\", """\\""")
+    fun applyJsonEscapes(string: String, format: OutputFormat):String{
+        val ret = string
+                .replace("\\", "\\\\")
                 .replace("\"", "\\\"")
+                .replace("\n", "\\n")
+                .replace("\r", "\\r")
+                .replace("\b", "\\b")
+                .replace("\t", "\\t")
 
+        return ret
+        /*if(format.isKotlin()){
+            return ret//.replace("\$", "\$")
+        }
+        else return ret
 
+         */
     }
 
-    fun applyUriEscapes(string: String, format: OutputFormat = OutputFormat.JAVA_JUNIT_4):String{
-        val ret = if(format.isKotlin()) string.replace("\$", "%24")
-        else string
-        return ret.replace("\\", "%5C")
+    fun applyUriEscapes(string: String, format: OutputFormat):String{
+        val ret = string.replace("\\", "%5C")
+                .replace("\"", "%22")
+                .replace("\n", "%0A")
+
+        if(format.isKotlin()) return ret.replace("\$", "%24")
+        else return ret
     }
```

Comment:
as a I said before, why not using the existing functions to escape URIs?

Reply:
I tried that and got some trouble, but it may be due to another bug overlapping it. I'll have another look at it.